pkg/planner/core/expression_rewriter.go

```diff
@@ -161,7 +162,6 @@ func buildSimpleExpr(ctx expression.BuildContext, node ast.ExprNode, opts ...exp
 	if rewriter.schema == nil {
 		rewriter.schema = expression.NewSchema()
 	}
-	rewriter.sctx.SetValue(expression.TiDBDecodeKeyFunctionKey, decodeKeyFromString)
```

Comment:
We use a static method to access decode key function: https://github.com/pingcap/tidb/pull/52015/files#diff-168c33d16a6ad85e67cafc2db1a37a3b5e8f10dfb5333ff98712954117069020R1357 which is easier to maintainand    expression context does not need to expo